lib/presentation/widget/myspace/my_space_widget.dart

```diff
@@ -321,51 +324,20 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
     return Container();
   }
 
-  void settingModalBottomSheet(BuildContext context, Document document) {
-    showModalBottomSheet(
-      shape: RoundedRectangleBorder(
-        borderRadius: BorderRadius.only(
-          topLeft: Radius.circular(20.0),
-          topRight: Radius.circular(20.0)
-        )),
-      context: context,
-      builder: (BuildContext buildContext) {
-        return Container(
-          child: Wrap(
-            children: <Widget>[
-              ListTile(
-              leading: SvgPicture.asset(
-                document.mediaType.getFileTypeImagePath(imagePath),
-                width: 16,
-                height: 20,
-                fit: BoxFit.fill,
-              ),
-              title: Transform(
-                transform: Matrix4.translationValues(-16, -2, 0.0),
-                child: Text(
-                  document.name,
-                  maxLines: 1,
-                  style: TextStyle(fontSize: 14, color: AppColor.documentNameItemTextColor),
-                ),
-              ),
-              trailing: Text(filesize(document.size),
-                style: TextStyle(fontSize: 14, color: AppColor.documentModifiedDateItemTextColor))),
-              Divider(),
-              ListTile(
-                leading: SvgPicture.asset(
-                  imagePath.icFileDownload,
-                  width: 24,
-                  height: 24,
-                  fit: BoxFit.fill),
-                title: Text(
-                  AppLocalizations.of(context).download_to_device,
-                  style: TextStyle(fontSize: 16, color: AppColor.documentNameItemTextColor)),
-                onTap: () => mySpaceViewModel.downloadFile(document.documentId)
-              ),
-            ],
-          ),
-        );
-      }
-    );
+  List<Widget> contextMenuActionTiles(Document document) {
+    return [
+      ListTile(
+        leading: SvgPicture.asset(
+          imagePath.icFileDownload,
+          width: 24,
+          height: 24,
+          fit: BoxFit.fill),
+        title: Text(
+          AppLocalizations.of(context).download_to_device,
+          style: TextStyle(
+            fontSize: 16,
+            color: AppColor.documentNameItemTextColor)),
+        onTap: () => mySpaceViewModel.downloadFile(document.documentId))
+    ];
```

Comment:
How about other action: `Details`, `Copy`, ... ? We need to duplicate the same of code for each item?
Please refer to my builders in login widget. `InputDecorationBuilder`, `LoginTextBuilder`, ...
It is an option, you can make your idea.

Reply:
What do you think of https://github.com/linagora/linshare-mobile-flutter-app/pull/13/commits/ecfdcb3093e1f1e53258d15949320c40ae400d85